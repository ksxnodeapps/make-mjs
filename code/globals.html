<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>@make-mjs/code — References</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.js" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">@make-mjs/code — References</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-only-exported" />
							<label class="tsd-widget" for="tsd-filter-only-exported">Only exported</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1> @make-mjs/code — References</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Modules</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-module"><a href="modules/importstatement.html" class="tsd-kind-icon">Import<wbr>Statement</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Interfaces</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-interface"><a href="interfaces/codetransformoptions.html" class="tsd-kind-icon">Code<wbr>Transform<wbr>Options</a></li>
								<li class="tsd-kind-interface tsd-has-type-parameter"><a href="interfaces/nodepathvisitor.html" class="tsd-kind-icon">Node<wbr>Path<wbr>Visitor</a></li>
								<li class="tsd-kind-interface"><a href="interfaces/nodetransformoptions.html" class="tsd-kind-icon">Node<wbr>Transform<wbr>Options</a></li>
								<li class="tsd-kind-interface"><a href="interfaces/reformatoptions.html" class="tsd-kind-icon">Reformat<wbr>Options</a></li>
								<li class="tsd-kind-interface"><a href="interfaces/traversaloptions.html" class="tsd-kind-icon">Traversal<wbr>Options</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Variables</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-variable"><a href="globals.html#generatecode" class="tsd-kind-icon">generate<wbr>Code</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Functions</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-function"><a href="globals.html#formatcode" class="tsd-kind-icon">format<wbr>Code</a></li>
								<li class="tsd-kind-function"><a href="globals.html#parsecode" class="tsd-kind-icon">parse<wbr>Code</a></li>
								<li class="tsd-kind-function"><a href="globals.html#transformcode" class="tsd-kind-icon">transform<wbr>Code</a></li>
								<li class="tsd-kind-function"><a href="globals.html#transformnode" class="tsd-kind-icon">transform<wbr>Node</a></li>
								<li class="tsd-kind-function"><a href="globals.html#traverseimportstatements" class="tsd-kind-icon">traverse<wbr>Import<wbr>Statements</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Object literals</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-object-literal"><a href="globals.html#default_parser_options" class="tsd-kind-icon">DEFAULT_<wbr>PARSER_<wbr>OPTIONS</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Variables</h2>
				<section class="tsd-panel tsd-member tsd-kind-variable">
					<a name="generatecode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> generate<wbr>Code</h3>
					<div class="tsd-signature tsd-kind-icon">generate<wbr>Code<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">generate</span><span class="tsd-signature-symbol"> =&nbsp;generate</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/generate-code.ts#L4">lib/generate-code.ts:4</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Functions</h2>
				<section class="tsd-panel tsd-member tsd-kind-function">
					<a name="formatcode" class="tsd-anchor"></a>
					<h3>format<wbr>Code</h3>
					<ul class="tsd-signatures tsd-kind-function">
						<li class="tsd-signature tsd-kind-icon">format<wbr>Code<span class="tsd-signature-symbol">(</span>code<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, options<span class="tsd-signature-symbol">?: </span><a href="interfaces/reformatoptions.html" class="tsd-signature-type">ReformatOptions</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/format-code.ts#L9">lib/format-code.ts:9</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>code: <span class="tsd-signature-type">string</span></h5>
								</li>
								<li>
									<h5><span class="tsd-flag ts-flagDefault value">Default value</span> options: <a href="interfaces/reformatoptions.html" class="tsd-signature-type">ReformatOptions</a><span class="tsd-signature-symbol"> =&nbsp;{}</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">string</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function">
					<a name="parsecode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> parse<wbr>Code</h3>
					<ul class="tsd-signatures tsd-kind-function">
						<li class="tsd-signature tsd-kind-icon">parse<wbr>Code<span class="tsd-signature-symbol">(</span>code<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, options<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">ParserOptions</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">File</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/parse-code.ts#L5">lib/parse-code.ts:5</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>code: <span class="tsd-signature-type">string</span></h5>
								</li>
								<li>
									<h5><span class="tsd-flag ts-flagDefault value">Default value</span> options: <span class="tsd-signature-type">ParserOptions</span><span class="tsd-signature-symbol"> =&nbsp;OPTIONS</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">File</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function">
					<a name="transformcode" class="tsd-anchor"></a>
					<h3>transform<wbr>Code</h3>
					<ul class="tsd-signatures tsd-kind-function">
						<li class="tsd-signature tsd-kind-icon">transform<wbr>Code<span class="tsd-signature-symbol">(</span>code<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, options<span class="tsd-signature-symbol">: </span><a href="interfaces/codetransformoptions.html" class="tsd-signature-type">CodeTransformOptions</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">GeneratorResult</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/transform.ts#L51">lib/transform.ts:51</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>code: <span class="tsd-signature-type">string</span></h5>
								</li>
								<li>
									<h5>options: <a href="interfaces/codetransformoptions.html" class="tsd-signature-type">CodeTransformOptions</a></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">GeneratorResult</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function">
					<a name="transformnode" class="tsd-anchor"></a>
					<h3>transform<wbr>Node</h3>
					<ul class="tsd-signatures tsd-kind-function">
						<li class="tsd-signature tsd-kind-icon">transform<wbr>Node<span class="tsd-signature-symbol">(</span>node<span class="tsd-signature-symbol">: </span><a href="" class="tsd-signature-type">Node</a>, options<span class="tsd-signature-symbol">: </span><a href="interfaces/nodetransformoptions.html" class="tsd-signature-type">NodeTransformOptions</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/transform.ts#L14">lib/transform.ts:14</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>node: <a href="" class="tsd-signature-type">Node</a></h5>
								</li>
								<li>
									<h5>options: <a href="interfaces/nodetransformoptions.html" class="tsd-signature-type">NodeTransformOptions</a></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function">
					<a name="traverseimportstatements" class="tsd-anchor"></a>
					<h3>traverse<wbr>Import<wbr>Statements</h3>
					<ul class="tsd-signatures tsd-kind-function">
						<li class="tsd-signature tsd-kind-icon">traverse<wbr>Import<wbr>Statements<span class="tsd-signature-symbol">(</span>node<span class="tsd-signature-symbol">: </span><a href="" class="tsd-signature-type">Node</a>, options<span class="tsd-signature-symbol">: </span><a href="interfaces/traversaloptions.html" class="tsd-signature-type">TraversalOptions</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">void</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/traverse-import-statements.ts#L40">lib/traverse-import-statements.ts:40</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>node: <a href="" class="tsd-signature-type">Node</a></h5>
								</li>
								<li>
									<h5>options: <a href="interfaces/traversaloptions.html" class="tsd-signature-type">TraversalOptions</a></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">void</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Object literals</h2>
				<section class="tsd-panel tsd-member tsd-kind-object-literal">
					<a name="default_parser_options" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> DEFAULT_<wbr>PARSER_<wbr>OPTIONS</h3>
					<div class="tsd-signature tsd-kind-icon">DEFAULT_<wbr>PARSER_<wbr>OPTIONS<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">object</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/default-parser-options.ts#L3">lib/default-parser-options.ts:3</a></li>
						</ul>
					</aside>
					<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-object-literal">
						<a name="default_parser_options.plugins" class="tsd-anchor"></a>
						<h3>plugins</h3>
						<div class="tsd-signature tsd-kind-icon">plugins<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol">(</span><span class="tsd-signature-type">"dynamicImport"</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">"importMeta"</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol"> =&nbsp;[&#x27;dynamicImport&#x27;, &#x27;importMeta&#x27;]</span></div>
						<aside class="tsd-sources">
							<ul>
								<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/default-parser-options.ts#L5">lib/default-parser-options.ts:5</a></li>
							</ul>
						</aside>
					</section>
					<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-object-literal">
						<a name="default_parser_options.sourcetype" class="tsd-anchor"></a>
						<h3>source<wbr>Type</h3>
						<div class="tsd-signature tsd-kind-icon">source<wbr>Type<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">"unambiguous"</span><span class="tsd-signature-symbol"> =&nbsp;&quot;unambiguous&quot;</span></div>
						<aside class="tsd-sources">
							<ul>
								<li>Defined in <a href="https://github.com/ksxnodeapps/make-mjs/blob/dca20b9/packages/code/lib/default-parser-options.ts#L4">lib/default-parser-options.ts:4</a></li>
							</ul>
						</aside>
					</section>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals current ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
					<li class=" tsd-kind-module">
						<a href="modules/importstatement.html">Import<wbr>Statement</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-interface">
						<a href="interfaces/codetransformoptions.html" class="tsd-kind-icon">Code<wbr>Transform<wbr>Options</a>
					</li>
					<li class=" tsd-kind-interface tsd-has-type-parameter">
						<a href="interfaces/nodepathvisitor.html" class="tsd-kind-icon">Node<wbr>Path<wbr>Visitor</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/nodetransformoptions.html" class="tsd-kind-icon">Node<wbr>Transform<wbr>Options</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/reformatoptions.html" class="tsd-kind-icon">Reformat<wbr>Options</a>
					</li>
					<li class=" tsd-kind-interface">
						<a href="interfaces/traversaloptions.html" class="tsd-kind-icon">Traversal<wbr>Options</a>
					</li>
					<li class=" tsd-kind-variable">
						<a href="globals.html#generatecode" class="tsd-kind-icon">generate<wbr>Code</a>
					</li>
					<li class=" tsd-kind-function">
						<a href="globals.html#formatcode" class="tsd-kind-icon">format<wbr>Code</a>
					</li>
					<li class=" tsd-kind-function">
						<a href="globals.html#parsecode" class="tsd-kind-icon">parse<wbr>Code</a>
					</li>
					<li class=" tsd-kind-function">
						<a href="globals.html#transformcode" class="tsd-kind-icon">transform<wbr>Code</a>
					</li>
					<li class=" tsd-kind-function">
						<a href="globals.html#transformnode" class="tsd-kind-icon">transform<wbr>Node</a>
					</li>
					<li class=" tsd-kind-function">
						<a href="globals.html#traverseimportstatements" class="tsd-kind-icon">traverse<wbr>Import<wbr>Statements</a>
					</li>
					<li class=" tsd-kind-object-literal">
						<a href="globals.html#default_parser_options" class="tsd-kind-icon">DEFAULT_<wbr>PARSER_<wbr>OPTIONS</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-module"><span class="tsd-kind-icon">Module</span></li>
				<li class="tsd-kind-object-literal"><span class="tsd-kind-icon">Object literal</span></li>
				<li class="tsd-kind-variable"><span class="tsd-kind-icon">Variable</span></li>
				<li class="tsd-kind-function"><span class="tsd-kind-icon">Function</span></li>
				<li class="tsd-kind-function tsd-has-type-parameter"><span class="tsd-kind-icon">Function with type parameter</span></li>
				<li class="tsd-kind-index-signature"><span class="tsd-kind-icon">Index signature</span></li>
				<li class="tsd-kind-type-alias"><span class="tsd-kind-icon">Type alias</span></li>
				<li class="tsd-kind-type-alias tsd-has-type-parameter"><span class="tsd-kind-icon">Type alias with type parameter</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-enum"><span class="tsd-kind-icon">Enumeration</span></li>
				<li class="tsd-kind-enum-member"><span class="tsd-kind-icon">Enumeration member</span></li>
				<li class="tsd-kind-property tsd-parent-kind-enum"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-enum"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-interface"><span class="tsd-kind-icon">Interface</span></li>
				<li class="tsd-kind-interface tsd-has-type-parameter"><span class="tsd-kind-icon">Interface with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-interface"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-class"><span class="tsd-kind-icon">Class</span></li>
				<li class="tsd-kind-class tsd-has-type-parameter"><span class="tsd-kind-icon">Class with type parameter</span></li>
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class"><span class="tsd-kind-icon">Accessor</span></li>
				<li class="tsd-kind-index-signature tsd-parent-kind-class"><span class="tsd-kind-icon">Index signature</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li>
				<li class="tsd-kind-accessor tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private accessor</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-call-signature tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
<script>if (location.protocol == 'file:') document.write('<script src="assets/js/search.js"><' + '/script>');</script>
</body>
</html>