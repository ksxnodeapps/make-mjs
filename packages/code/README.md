# @make-mjs/code

Turn a Babel JavaScript code into a MJS code

## Example

Let's say you have `./lib/foo.js` and `./lib/bar.js` and this file (named `index.js`):

```javascript
export * from './lib/foo'
export * from './lib/bar'
```

You run this command:

```typescript
import fs from 'fs'
import transformCode from '@make-mjs/code'
const input = fs.readFileSync('index.js', 'utf8')
const output = transformCode(code, { moduleContainer: ['node_modules'] }).code
fs.writeFileSync('index.mjs', output)
```

You should get this file (named `index.mjs`):

```javascript
export * from './lib/foo.mjs'
export * from './lib/bar.mjs'
```

## License

[MIT](https://git.io/JeY5b) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
