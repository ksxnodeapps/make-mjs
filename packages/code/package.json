{
  "name": "@make-mjs/code",
  "version": "0.2.7",
  "description": "Turn a Babel JavaScript file into a MJS code",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://ksxnodeapps.github.io/make-mjs/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ksxnodeapps/make-mjs.git"
  },
  "bugs": {
    "url": "https://github.com/ksxnodeapps/make-mjs/issues"
  },
  "keywords": [
    "es-module",
    "babel",
    "import",
    "esm",
    "mjs"
  ],
  "sideEffects": false,
  "dependencies": {
    "fs-extra": "^8.1.0",
    "@babel/types": "^7.7.4",
    "@babel/parser": "^7.7.7",
    "@babel/generator": "^7.7.7",
    "@babel/traverse": "^7.7.4",
    "remote-controlled-promise": "^0.0.6",
    "@tsfun/object": "^0.0.7",
    "tslib": "^1.10.0",
    "@make-mjs/url": "^0.2.7",
    "@types/babel__generator": "^7.6.1",
    "@types/babel__traverse": "^7.0.8",
    "@types/fs-extra": "^8.0.1",
    "@types/node": "^13.1.6"
  }
}
