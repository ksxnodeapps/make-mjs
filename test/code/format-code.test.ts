import { formatCode } from '@make-mjs/code'

it('simple use case', () => {
  const code = `
    import a from 'a'
    import b from 'b'
    import c from 'c'
    console.log(a)
    if (b) c()
    import('d').then(console.log)
  `

  expect(formatCode(code)).toMatchSnapshot()
})
