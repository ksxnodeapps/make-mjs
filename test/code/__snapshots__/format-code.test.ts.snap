// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple use case 1`] = `
"import a from 'a';
import b from 'b';
import c from 'c';
console.log(a);
if (b) c();
import('d').then(console.log);"
`;
